tree/ntuple/v7/src/RFieldMeta.cxx

```diff
@@ -186,39 +251,145 @@ std::size_t ROOT::Experimental::RClassField::AppendImpl(const void *from)
 
 void ROOT::Experimental::RClassField::ReadGlobalImpl(NTupleSize_t globalIndex, void *to)
 {
+   for (const auto &[_, si] : fStagingItems) {
+      CallReadOn(*si.fField, globalIndex, fStagingArea.get() + si.fOffset);
+   }
    for (unsigned i = 0; i < fSubFields.size(); i++) {
       CallReadOn(*fSubFields[i], globalIndex, static_cast<unsigned char *>(to) + fSubFieldsInfo[i].fOffset);
    }
 }
 
 void ROOT::Experimental::RClassField::ReadInClusterImpl(RNTupleLocalIndex localIndex, void *to)
 {
+   for (const auto &[_, si] : fStagingItems) {
+      CallReadOn(*si.fField, localIndex, fStagingArea.get() + si.fOffset);
+   }
    for (unsigned i = 0; i < fSubFields.size(); i++) {
       CallReadOn(*fSubFields[i], localIndex, static_cast<unsigned char *>(to) + fSubFieldsInfo[i].fOffset);
    }
 }
 
-void ROOT::Experimental::RClassField::BeforeConnectPageSource(Internal::RPageSource &pageSource)
+ROOT::Experimental::DescriptorId_t ROOT::Experimental::RClassField::LookupMember(const RNTupleDescriptor &desc,
+                                                                                 std::string_view memberName,
+                                                                                 DescriptorId_t classFieldId)
 {
-   // This can happen for added base classes or non-simple members.
-   if (GetOnDiskId() == kInvalidDescriptorId) {
-      return;
+   auto idSourceMember = desc.FindFieldId(memberName, classFieldId);
+   if (idSourceMember != kInvalidDescriptorId)
+      return idSourceMember;
+
+   for (const auto &subFieldDesc : desc.GetFieldIterable(classFieldId)) {
+      const auto subFieldName = subFieldDesc.GetFieldName();
+      if (subFieldName.length() > 2 && subFieldName[0] == ':' && subFieldName[1] == '_') {
+         idSourceMember = LookupMember(desc, memberName, subFieldDesc.GetId());
+         if (idSourceMember != kInvalidDescriptorId)
+            return idSourceMember;
+      }
+   }
+
+   return kInvalidDescriptorId;
+}
+
+void ROOT::Experimental::RClassField::SetStagingClass(const std::string &className, unsigned int classVersion)
+{
+   TClass::GetClass(className.c_str())->GetStreamerInfo(classVersion);
+   if (classVersion != GetTypeVersion()) {
+      fStagingClass = TClass::GetClass((className + std::string("@@") + std::to_string(classVersion)).c_str());
```

Comment:
For the search based on checksum search case, we don't know the version use for the name.  Instead we need to use something along the line of:
```
auto info = TClass::GetClass(className.c_str())->GetStreamerInfo(classVersion);
auto elem = info->GetElement(0);
if (elem && elem->GetType() == TStreamerInfo::kCacheNew)
   fStagingClass = elem->GetClass();
```
which will work for both.

Reply:
Will be addressed in subsequent PR, see https://github.com/root-project/root/issues/17713